File: solutions/typescript/difference-of-squares/1/difference-of-squares.ts
export class Squares {
  private count: number
  constructor(count: number) {
    this.count = count;
  }

  get sumOfSquares(): number {
    let ans = 0
    for (let i = 1; i <= this.count; i++) {
      ans += i**2
    }
    return ans
  }

  get squareOfSum(): number {
    let ans = 0
    for (let i = 1; i <= this.count; i++) {
      ans += i
    }
    return ans**2
  }

  get difference(): number {
    return this.squareOfSum - this.sumOfSquares
  }
}

File: solutions/kotlin/difference-of-squares/1/src/main/kotlin/DifferenceOfSquares.kt
class Squares(val n: Int) {
    //TODO: implement proper constructor
    

    fun sumOfSquares(): Int {
        var ans = 0
        for (i in 0..n) {
            ans += i*i
        }
        return ans
    }

    fun squareOfSum(): Int {
        var ans = 0
        for (i in 0..n) {
            ans += i
        }
        ans *= ans
        return ans
    }

    fun difference(): Int {
        return squareOfSum() - sumOfSquares()
    }
}
